c_cxx/OpenVINO_EP/Linux/squeezenet_classification/squeezenet_cpp_app_io.cpp

```diff
@@ -0,0 +1,429 @@
+    resizedImageRGB.convertTo(resizedImage, CV_32F, 1.0 / 255);
+
+    // step 5: Split the RGB channels from the image.   
+    cv::Mat channels[3];
+    cv::split(resizedImage, channels);
+
+    //step 6: Normalize each channel.
+    // Normalization per channel
+    // Normalization parameters obtained from
+    // https://github.com/onnx/models/tree/master/vision/classification/squeezenet
+    channels[0] = (channels[0] - 0.485) / 0.229;
+    channels[1] = (channels[1] - 0.456) / 0.224;
+    channels[2] = (channels[2] - 0.406) / 0.225;
+
+    //step 7: Merge the RGB channels back to the image.
+    cv::merge(channels, 3, resizedImage);
+
+    // step 8: Convert the image to CHW RGB float format.
+    // HWC to CHW
+    cv::dnn::blobFromImage(resizedImage, preprocessedImage);
+    
+    //Run Inference
+    std::vector<const char*> inputNames{inputName};
+    std::vector<const char*> outputNames{outputName};
+
+    /* To run inference using ONNX Runtime, the user is responsible for creating and managing the 
+    input and output buffers. The buffers are IO Binding Buffers created on Remote Folders to create Remote Blob*/
+
+    size_t inputTensorSize = vectorProduct(inputDims);
+    std::vector<float> inputTensorValues(inputTensorSize);
+    inputTensorValues.assign(preprocessedImage.begin<float>(),
+                             preprocessedImage.end<float>());
+
+    size_t imgSize = inputTensorSize*4;
+    cl_int err;
+    cl::Buffer shared_buffer(ocl_instance->_context, CL_MEM_READ_WRITE, imgSize, NULL, &err);
+    {
+        void *buffer = (void *)preprocessedImage.ptr();
+        ocl_instance->_queue.enqueueWriteBuffer(shared_buffer, true, 0, imgSize, buffer);
+    }
+
+    //To pass to OrtValue wrap the buffer in shared buffer
+    void *shared_buffer_void = static_cast<void *>(&shared_buffer);
+    size_t outputTensorSize = vectorProduct(outputDims);
+    std::vector<float> outputTensorValues(outputTensorSize);
+ 
+    Ort::Value inputTensors = Ort::Value::CreateTensor(
+        info_gpu, shared_buffer_void, imgSize, inputDims.data(),
+        inputDims.size(), ONNX_TENSOR_ELEMENT_DATA_TYPE_FLOAT);
+
+    assert(("Output tensor size should equal to the label set size.",
+            labels.size() == outputTensorSize));
+    
+    size_t imgSizeO = outputTensorSize*4;
+    cl::Buffer shared_buffer_out(ocl_instance->_context, CL_MEM_READ_WRITE, imgSizeO, NULL, &err);
+   
+   //To pass the ORT Value wrap the output buffer in shared buffer
+    void *shared_buffer_out_void = static_cast<void *>(&shared_buffer_out);
+       Ort::Value outputTensors = Ort::Value::CreateTensor(
+        info_gpu, shared_buffer_out_void, imgSizeO, outputDims.data(),
+        outputDims.size(), ONNX_TENSOR_ELEMENT_DATA_TYPE_FLOAT);
+
+    /*------------------------------------------------------------------------*/
+  
+    /*If you want to create CPU Folders
+    
+    Ort::MemoryInfo memoryInfo = Ort::MemoryInfo::CreateCpu(
```

Comment:
@sfatimar @mayavijx Is this change required, which is commented out here. 
so, the code in the comment indicates, if they want to use this feature for CPU?

Reply:
You can keep CPU Folders and it will work. What will not work is Remote Buffers with Raw Device Pointers and Running without IOBuffering. We do not have support to copy From Remote Buffers to CPU

Reply:
Updated the script by removing the commented section.